test/treiber_stack/dune

```diff
@@ -7,6 +7,11 @@
 (test
  (name treiber_stack_dscheck)
  (libraries atomic dscheck alcotest)
+ (enabled_if
+  (not
+   (and
+    (= %{arch_sixtyfour} false)
+    (= %{architecture} arm))))
```

Comment:
Why some datastructures need that, but not the `spsc` queue and the `mpsc` queue ?

Reply:
Some tests are quite long running and cause the CI to time out. I've only disabled those.